### PR TITLE
Remove `dataclass` from `_MultiTensor` and `TensorFrame`

### PR DESCRIPTION
**Pull request by weihua916**

### CODE REVIEW

File: torch_frame/data/multi_nested_tensor.py [hunk 1]
```diff
@@ -235,21 +244,27 @@ def _row_narrow(self, start: int, length: int) -> 'MultiNestedTensor':
...
+    def _col_narrow(self, start: int, length: int) -> "MultiNestedTensor":
         r"""Helper function called by :obj:`narrow`."""
         assert start >= 0
         assert length > 0
         end = start + length
         if start == 0:
             assert end < self.num_cols
-            offset_mat = (self.offset[:-1].reshape(
-                self.num_rows, self.num_cols)[:, start:end + 1])
+            offset_mat = self.offset[:-1].reshape(self.num_rows,
```
Comment (rusty1s) [inline torch_frame/data/multi_nested_tensor.py:261]: Formatting is off. I recommend to split into two commands.

**Review by rusty1s — APPROVED** [review]
What‘s up with the formatting?

File: torch_frame/data/multi_embedding_tensor.py [hunk 2]
```diff
@@ -36,6 +36,15 @@ class MultiEmbeddingTensor(_MultiTensor):
         >>> out[0, 2]
         tensor([10])
     """
+    def __init___(
```
[the remark this answers is not in the document]
Comment (weihua916) [reply torch_frame/data/multi_embedding_tensor.py:39]: Ah I see. thanks.

File: torch_frame/data/multi_nested_tensor.py (hunk 1 above)
Comment (weihua916) [reply torch_frame/data/multi_nested_tensor.py:261]: Fixed!

**Comment by weihua916** [comment]
> What‘s up with the formatting?

I started using `"editor.defaultFormatter": "charliermarsh.ruff"` in my VScode because we shifted to `ruff` in https://github.com/pyg-team/pytorch-frame/pull/205

**Comment by akihironitta** [comment]
@weihua916 @rusty1s I'm just curious 📝, could I ask what the motivation behind this change was?

**Pull request merged**